tests/unit/tasks/models/jobsModel.spec.ts

```diff
@@ -0,0 +1,363 @@
+
+      expect(getJobStatusMock).toHaveBeenCalledTimes(1);
+      expect(publishToCatalogMock).toHaveBeenCalledTimes(1);
+      expect(publishLayerMock).toHaveBeenCalledTimes(1);
+
+      expect(triggerSyncMock).toHaveBeenCalledTimes(1);
+      const mapPublishReqForRasterMap = { ...mapPublishReq };
+      mapPublishReqForRasterMap.name = `test-${rasterMapTestData.productType}`;
+      mapPublishReqForRasterMap.tilesPath = `test/${rasterMapTestData.productType}`;
+      expect(publishLayerMock).toHaveBeenCalledWith(mapPublishReqForRasterMap);
+
+      const expectedPublishTocCatalogReq = { ...catalogReqData };
+      expectedPublishTocCatalogReq.metadata.productType = ProductType.RASTER_MAP;
+      expect(publishToCatalogMock).toHaveBeenCalledWith(expectedPublishTocCatalogReq);
+      expect(triggerSyncMock).toHaveBeenCalledWith('test', '1', ProductType.RASTER_MAP, OperationTypeEnum.ADD, mapPublishReqForRasterMap.tilesPath);
+    });
+
+    it('do nothing if some tasks are not done', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ mapServerCacheType: 'fs', 'tiling.zoomGroups': '' });
+      setValue('ingestionNewJobType', ingestionNewJobType);
+      setValue('ingestionTaskType', { tileMergeTask, tileSplitTask });
+
+      getJobStatusMock.mockReturnValue({
+        allCompleted: false,
+        type: ingestionNewJobType,
+      });
+
+      getTaskMock.mockReturnValue({
+        type: tileSplitTask,
+      });
+
+      // tasksManager = new TasksManager(
+      //   logger,
+      //   configMock,
+      //   syncClientMock,
+      //   jobManagerClientMock,
+      //   mapPublisherClientMock,
+      //   catalogClientMock,
+      //   linkBuilderMock,
+      //   metadataMergerMock
+      // );
+
+      await jobsManager.completeJob(jobId, taskId);
+
+      expect(getJobStatusMock).toHaveBeenCalledTimes(1);
+      expect(publishToCatalogMock).toHaveBeenCalledTimes(0);
+      expect(publishLayerMock).toHaveBeenCalledTimes(0);
+      expect(triggerSyncMock).toHaveBeenCalledTimes(0);
+    });
+
+    it('should abort all merge tasks and make job failed on task failure', async function () {
+      setValue('ingestionUpdateJobType', ingestionUpdateJobType);
+      setValue('ingestionTaskType', { tileMergeTask, tileSplitTask });
+
+      const rasterMapTestData = { ...testMetadata };
+      rasterMapTestData.productType = ProductType.RASTER_MAP;
+
+      getJobStatusMock.mockReturnValue({
+        id: jobId,
+        completed: true,
+        successful: true,
+        relativePath: `test/${ProductType.RASTER_MAP}`,
+        percentage: 80,
+        metadata: rasterMapTestData,
+        type: ingestionUpdateJobType,
+        status: OperationStatus.IN_PROGRESS,
+      });
+
+      getTaskMock.mockReturnValue({
+        id: taskId,
+        jobId: jobId,
+        type: tileMergeTask,
+        status: OperationStatus.FAILED,
+      });
+
+      // tasksManager = new TasksManager(
```

Comment:
why the remarks?

Reply:
Pending

should have been deleted, i removed it, old test initialized task manager every test, i made it initialize once before all and its called job manager now instad of task manager , see old tests => https://github.com/MapColonies/discrete-overseer-creator/blob/master/tests/unit/tasks/models/tasksModel.spec.ts